client/lib/plans/index.js

```diff
@@ -222,25 +227,33 @@ export function applyTestFiltersToPlansList( planName ) {
 	const filteredPlanConstantObj = { ...plansList[ planName ] };
 	let filteredPlanFeaturesConstantObj = pick( featuresList, plansList[ planName ].getFeatures() );
 
+	// remove disabled features from list
 	if ( ! isWordadsInstantActivationEnabled() ) {
 		filteredPlanFeaturesConstantObj = pickBy( filteredPlanFeaturesConstantObj,
 			( value, key ) => key !== WORDADS_INSTANT
 		);
-	} else if ( planName === PLAN_PREMIUM ) {
-		filteredPlanConstantObj.getDescription = plansList[ planName ].getDescriptionWithWordAdsInstantActivation;
 	}
-
 	if ( ! isGoogleVouchersEnabled() ) {
 		filteredPlanFeaturesConstantObj = pickBy( filteredPlanFeaturesConstantObj,
 			( value, key ) => key !== FEATURE_GOOGLE_AD_CREDITS
 		);
-	} else if ( planName === PLAN_PREMIUM ) {
+	}
+
+	// update plan descriptions
+	if ( isWordadsInstantActivationEnabled() && isGoogleVouchersEnabled() && planName === PLAN_PREMIUM ) {
+		filteredPlanConstantObj.getDescription = plansList[ planName ].getDescriptionWithWordAdsAndGoogleVouchers;
+	} else if ( isWordadsInstantActivationEnabled() && planName === PLAN_PREMIUM ) {
+		filteredPlanConstantObj.getDescription = plansList[ planName ].getDescriptionWithWordAdsInstantActivation;
+	} else if ( isGoogleVouchersEnabled() && planName === PLAN_PREMIUM ) {
 		filteredPlanConstantObj.getDescription = plansList[ planName ].getDescriptionWithGoogleVouchers;
-	} else if ( isWordpressAdCreditsEnabled() && planName === PLAN_BUSINESS	) {
+	} else if ( isWordpressAdCreditsEnabled() && planName === PLAN_BUSINESS ) {
 		filteredPlanConstantObj.getDescription = plansList[ planName ].getDescriptionWithWordAdsCredit;
+	}
 
+	// update info popup
+	if ( isWordpressAdCreditsEnabled() && planName === PLAN_BUSINESS	) {
 		filteredPlanFeaturesConstantObj[ FEATURE_GOOGLE_AD_CREDITS ].getDescription =
-			featuresList[ FEATURE_GOOGLE_AD_CREDITS ].getDescriptionWithWordAdsCredit;
+		featuresList[ FEATURE_GOOGLE_AD_CREDITS ].getDescriptionWithWordAdsCredit;
 	}
```

Comment:
Thanks for making this fn easier to read. I think that we could upgrade it even more by moving the parts to arrow functions inside the `applyTestFiltersToPlansList` fn with proper names so we can get rid of the comments. What do you think? I think code > comments.